### PR TITLE
[SU-171] Remove Rawls implementation of GooglePubSubDAO, use workbench-libs instead

### PR DESCRIPTION
**Pull request by MatthewBemis**
I recommend reviewing https://github.com/broadinstitute/workbench-libs/pull/1040 first.

This PR removes a duplicate implementation of the GooglePubSubDAO. Rawls was the first service to use PubSub, so we created the class here. At some point later on, we copied that class to workbench-libs so other services could use it, but never removed the duplicate code from Rawls. This has resulted in divergent implementations. I've removed the duplicate implementation here, harmonized the workbench-libs GooglePubSubDAO implementation with what was in Rawls, and switched Rawls to use the workbench-libs version of GooglePubSubDAO.

Almost all of the changes in here are changes to imports and deleted code.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -93,7 +93,7 @@ object Dependencies {
   val workbenchLibsHash = "20f9225"
 
   val workbenchModelV  = s"0.15-${workbenchLibsHash}"
-  val workbenchGoogleV = s"0.21-${workbenchLibsHash}"
+  val workbenchGoogleV = s"0.21-805f2011-SNAP"
```
Comment (MatthewBemis) [inline project/Dependencies.scala]: I will migrate to a non-snap once the relevant workbench-libs PR is merged. There was a small bit of work to converge the Rawls and workbench-libs pub sub DAO implementations, as the Rawls one had progressed a bit over the years whereas workbench-libs stayed the same

**Comment by MatthewBemis** [comment]
jenkins retest

**Comment by davidangb** [comment]
I know this is still in draft so feel free to ignore me … but don't forget to remove the `google-api-services-pubsub` library from Dependencies!

File: project/Dependencies.scala [hunk 2]
```diff
@@ -85,7 +84,7 @@ object Dependencies {
   val workbenchLibsHash = "46d1df6"
 
   val workbenchModelV  = s"0.15-${workbenchLibsHash}"
-  val workbenchGoogleV = s"0.22-${workbenchLibsHash}"
+  val workbenchGoogleV = s"0.23-6466499a-SNAP"
```
Comment (MatthewBemis) [inline project/Dependencies.scala]: Will update to non-SNAP once https://github.com/broadinstitute/workbench-libs/pull/1040 is merged

**Review by nawatts — APPROVED** [review]
Excellent ratio of lines deleted to lines added.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 3]
```diff
@@ -281,16 +281,8 @@ object Boot extends IOApp with LazyLogging {
...
-        clientEmail,
-        pathToPem,
-        appName,
-        serviceProject,
-        workbenchMetricBaseName = metricsPrefix
-      )
-
       val notificationDAO = new PubSubNotificationDAO(
-        notificationPubSubDAO,
+        pubSubDAO,
```
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:289]: with the library change in this PR, could you also update the constructors for `pubSubDAO` and `importServicePubSubDAO` at line 187 to use the non-deprecated signature?

**Review by davidangb — APPROVED** [review]
one suggestion inline, and still need to update to the published version of workbench-libs after that merges, otherwise looks :+1:

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala (hunk 3 above)
Comment (MatthewBemis) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:289]: Done in 6ab019f10

**Pull request merged**